code/game/objects/structures/coatrack.dm

```diff
@@ -0,0 +1,124 @@
+/obj/structure/coatrack
+	name = "coat rack"
+	desc = "Rack that holds coats, or hats, if you're so inclined."
+	icon = 'icons/obj/coatrack.dmi'
+	icon_state = "coatrack"
+	var/obj/item/clothing/coat
+	var/obj/item/clothing/head/hat
+	var/list/custom_sprites = list(/obj/item/clothing/head/beret/security, /obj/item/clothing/accessory/poncho/tajarancloak) // Custom manual sprite override.
+
+/obj/structure/coatrack/attack_hand(mob/user as mob)
+	if(!ishuman(user))
+		return
+	if(user.incapacitated())
+		return
```

Comment:
You should use the use_check / use_check_and_message proc instead possibly with optional flags.